### DISCUSSION TITLE
error uploading a file with a post request

### DISCUSSION THREAD

**Discussion by fdennig** [fastapi/fastapi#7580] [Questions] [answered]
### Description
How do I upload a file with a post request?
I have done what is described here - https://fastapi.tiangolo.com/tutorial/request-files/ - so the following code is in my main.py file
```
from fastapi import FastAPI, Request, File, UploadFile

app = FastAPI

@app.post('/uploaddata/')
async def create_upload_file(file: UploadFile = File(...)):
    return {"filename": file.filename}
```
and on the client side i have a simple form like this:
```
  <form method="post" action="/uploaddata/" enctype="multipart/form-data">
            <button class="btn btn-outline-secondary" type="submit">upload</button>
            <input type="file" id="file_upload">
   </form>
```
I get the error: {"detail":"There was an error parsing the body"}, just like issue #201
https://github.com/tiangolo/fastapi/issues/201
In that case the problem seemed to be that python-multipart wasn't installed. I have installed it, but I still get the error. 
Is there something obviously wrong? If not, how can I get hold of more information that just this reply:
 {"detail":"There was an error parsing the body"}?

thanks!

**Comment by retnikt** [comment]
Can you please format the code using [markdown code blocks](https://help.github.com/en/github/writing-on-github/basic-writing-and-formatting-syntax#quoting-code). Right now we cannot see your HTML because it is automatically removed by the formatter.

**Comment by fdennig** [comment]
@retnikt sorry about that. I'm new to this. I just reformatted the code into blocks. 
thanks for any help you might provide!

**Comment by retnikt** [comment, marked as answer]
You are missing the `name` attribute on the HTML form, it must be the same as the parameter name (in your case `file`). So:
```html
<form method="post" action="/uploaddata/" enctype="multipart/form-data">
    <button class="btn btn-outline-secondary" type="submit">upload</button>
    <input type="file" id="file_upload" name="file">
</form>
```

**Comment by fdennig** [comment]
dammit that was easy. thanks @retnikt for the prompt and correct solution. 
could i ask one more stupid question? is it the case that every post request uploading a file must have a name matching the serverside parameter, or is this specific to fastapi?
thanks again, this is super helpful.

**Comment by retnikt** [comment]
This is a standard for all file uploads. See: https://tools.ietf.org/html/rfc7578 and https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/file

**Comment by fdennig** [comment]
thanks @retnikt

**Comment by tiangolo** [comment]
Thanks for the help here @retnikt ! :clap: :bow: 

Thanks for reporting back and closing the issue @fdennig  :+1:

**Comment by pranithan-kang** [comment]
This question links to this [issue](https://github.com/tiangolo/fastapi/issues/405).
